contracts/concert/src/lib.rs

```diff
@@ -254,13 +259,14 @@ impl From<Concert> for State {
             concert_id,
             running,
             metadata,
+            token_id,
         } = value;
 
-        let buyers = buyers.iter().copied().collect();
+        let buyers = buyers.into_iter().collect();
 
         let metadata = metadata
-            .iter()
-            .map(|(k, v)| (*k, v.iter().map(|(k, v)| (*k, v.clone())).collect()))
+            .into_iter()
+            .map(|(k, v)| (k, v.into_iter().map(|(k, v)| (k, v)).collect()))
```

Comment:
No need use map here
```suggestion
            .map(|(k, v)| (k, v.into_iter().collect()))
```